packages/protocol/contracts/L2/TaikoL2.sol

```diff
@@ -321,36 +321,38 @@ contract TaikoL2 is EssentialContract, TaikoL2Signer, ICrossChainSync {
     function _calcBasefee(
         EIP1559Config memory config,
         uint256 timeSinceParent,
-        uint32 gasLimit,
         uint32 parentGasUsed
     )
         private
         view
         returns (uint256 _basefee, uint64 _gasExcess)
     {
-        // Very important to cap _gasExcess uint64
-        unchecked {
-            uint32 parentGasUsedNet;
-            if (parentGasUsed > LibL2Consts.ANCHOR_GAS_COST) {
+        uint32 parentGasUsedNet;
+
+        if (parentGasUsed > LibL2Consts.ANCHOR_GAS_COST) {
+            unchecked {
                 parentGasUsedNet = parentGasUsed - LibL2Consts.ANCHOR_GAS_COST;
             }
-
-            uint256 a = uint256(gasExcess) + parentGasUsedNet;
-            uint256 b = config.gasIssuedPerSecond * timeSinceParent;
-            _gasExcess = uint64((a.max(b) - b).min(type(uint64).max));
         }
 
         _basefee = Lib1559Math.calculatePrice({
             xscale: config.xscale,
             yscale: config.yscale,
-            xExcess: _gasExcess,
-            xPurchase: gasLimit
+            xExcess: gasExcess,
+            xPurchase: parentGasUsedNet
```

Comment:
I think this adds an additional delay on the actual basefee that isn't really required, because this will set the basefee between the two points, but the parent block has already happened so the basefee needs to be at the end point `uint256(gasExcess) + parentGasUsedNet` already. I think the more accurate way to do this is to purchase 0 gas after already taking into account the parent gas ussed, which means calculating the derivative at the end point which is also easier/cheaper to calculate: `basefee = eth_qty(_gasExcess) / (TARGET * ADJUSTMENT_QUOTIENT)`.

The current implementation additionally followed an approach very similar to the one proposed in https://ethresear.ch/t/make-eip-1559-more-like-an-amm-curve/9082 in option 2, purchasing an extra amount of gas for the max amount of gas possible in the block for safety and accuracy, but without implementing the refund because that would be very complex to do.

Reply:
Check out this [new commit](https://github.com/taikoxyz/taiko-mono/pull/14388/commits/d9fbef87a145aab2fad84c1873dbd150f1568fad), let me know if it addresses your concerns.